internal/pkg/cli/pipeline_status.go

```diff
@@ -84,25 +83,30 @@ func newPipelineStatusOpts(vars pipelineStatusVars) (*pipelineStatusOpts, error)
 	}, nil
 }
 
-// Validate returns an error if the values provided by the user are invalid.
+// Validate returns an error if the optional flag values provided by the user are invalid.
 func (o *pipelineStatusOpts) Validate() error {
 	if o.appName != "" {
 		if _, err := o.store.GetApplication(o.appName); err != nil {
-			return err
-		}
-	}
-	if o.pipelineName != "" {
-		if _, err := o.pipelineSvc.GetPipeline(o.pipelineName); err != nil {
-			return err
+			return fmt.Errorf("validate application name: %w", err)
 		}
 	}
 	return nil
 }
 
-// Ask prompts for fields that are required but not passed in.
+// Ask prompts for fields that are required but not passed in, and validates those that are.
 func (o *pipelineStatusOpts) Ask() error {
-	if err := o.askAppName(); err != nil {
-		return err
+	if o.name != "" {
+		_, err := o.codepipeline.GetPipeline(o.name)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
```

Comment:
```suggestion
	if o.name != "" {
		if _, err := o.codepipeline.GetPipeline(o.name); err != nil {
			return err
		}
		return nil
	} else {
	  if err := o.askPipelineName(); err != nil {
	    return err
	  }
	}
```

Reply:
Are there benefits to doing it this way? I'd had it like this for `pipeline show`, but agreed with feedback that it may be nicer to get rid of the `else`: https://github.com/aws/copilot-cli/pull/3328#discussion_r820907333

What do you think?

Reply:
Oh i misread the code. I agree we can remove the `else` but it doesn't make sense to me to split the pipeline name flag check and the ask. Maybe we should check the app name first then the pipeline name.

Reply:
Oh, we did that because we only need the app name if we're asking for the pipeline name. If they pass in the pipeline name, we don't even need the app name. https://github.com/aws/copilot-cli/pull/3328#discussion_r819997029

Reply:
discussed offline. changed code for both `pipeline show` and `pipeline status` to require app name regardless of pipeline flag presence/absence.